aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -490,6 +490,20 @@ void dot<float>(CUDABLAS_DOT_ARGTYPES(float)) {
   TORCH_CUDABLAS_CHECK(cublasSdot(handle, n, x, incx, y, incy, result));
 }
 
+template <>
+void dot<c10::complex<double>>(CUDABLAS_DOT_ARGTYPES(c10::complex<double>)) {
+  TORCH_CUDABLAS_CHECK(cublasZdotu(handle, n, reinterpret_cast<const cuDoubleComplex*>(x),
```

Comment:
Hmm, I wonder if we shouldn't have some methods on `c10::complex` for doing pointery conversions like this. It would be nice to not have to be slinging reinterpret cast everywhere. (No action needed for PR)